eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/batch/io/active/AccessionedVariantMongoReader.java

```diff
@@ -76,45 +74,36 @@ public void open(ExecutionContext executionContext) throws ItemStreamException {
     }
 
     protected List<Bson> buildAggregation() {
-        Bson match = Aggregates.match(Filters.eq(REFERENCE_ASSEMBLY_FIELD, assemblyAccession));
+        Bson match = Aggregates.match(eq(REFERENCE_ASSEMBLY_FIELD, assemblyAccession));
         Bson sort = Aggregates.sort(orderBy(ascending(CONTIG_FIELD, START_FIELD)));
         Bson singlemap = Aggregates.match(Filters.not(exists(MAPPING_WEIGHT_FIELD)));
         List<Bson> aggregation = new ArrayList<>(Arrays.asList(match, sort, singlemap));
-        String tempArrayName = "ssArray";
+
         for (String submittedVariantCollectionName : allSubmittedVariantCollectionNames) {
-            Bson lookup = Aggregates.lookup(submittedVariantCollectionName, ACCESSION_FIELD,
-                                            CLUSTERED_VARIANT_ACCESSION_FIELD,
-                                            submittedVariantCollectionName);
+            String lookupQuery = "{ $lookup: { " +
+                    String.format("from: \"%s\",", submittedVariantCollectionName) +
+                    String.format("let: { rsAccession: \"$%s\" },", ACCESSION_FIELD) +
+                    "pipeline: [{" +
+                    "$match: {$expr: {$and: [" +
+                    String.format("{ $eq: ['$%s', \"$$rsAccession\"]},",
+                                  CLUSTERED_VARIANT_ACCESSION_FIELD) +
+                    String.format("{ $eq: [\"$%s\", \"%s\"]},",
+                                  REFERENCE_ASSEMBLY_FIELD_IN_SUBMITTED_COLLECTIONS,
+                                  this.assemblyAccession) +
+                    String.format("{ $eq: [\"$%s\", %d]}]}}}],",
+                                  TAXONOMY_FIELD, this.taxonomyAccession) +
+                    String.format("as: \"%s\"}}", submittedVariantCollectionName);
```

Comment:
Our Spring boot version [does not support](https://github.com/spring-projects/spring-data-mongodb/issues/3322) lookup with "let" keyword. So I had to resort to constructing a manual MongoDB query to make lookups with multiple join conditions. Unfortunately I won't be able to make the same optimiziations on other readers (merged reader, merged deprecated reader etc.,) because of a technicality: those lookups involve join conditions on array elements but indexes on array elements [are not supported](https://www.mongodb.com/docs/v4.0/core/index-multikey/#expr) by the expr keyword.